### PR TITLE
fix num classes sparse handling

### PR DESCRIPTION
**Pull request by imatiach-msft**
fix num classes sparse handling (used by explanation client)

EDIT: also fixes local explanation feature importance values for multiclass case

### CODE REVIEW

File: python/interpret_community/explanation/explanation.py [hunk 1]
```diff
@@ -338,8 +338,9 @@ def local_importance_values(self):
             the last will be for "fish". If you choose to pass in a classes array to the explainer, the names should
             be passed in using this same order.
         :rtype: list[list[float]] or list[list[list[float]]] or scipy.sparse.csr_matrix
+            or list[scipy.sparse.csr_matrix]
         """
-        if sp.sparse.issparse(self._local_importance_values):
+        if sp.sparse.issparse(self._local_importance_values) or isinstance(self._local_importance_values, list):
```
Comment (wamartin-aml) [inline python/interpret_community/explanation/explanation.py]: It's very weird that we're communicating sparse multiclass based on whether `_local_importance_values` is a list, and beyond being weird it's pretty fragile in the sense that at some point regular dense `_local_importance_values` could easily be a list. Can we have a field in the explanation keeping track?

File: python/interpret_community/explanation/explanation.py [hunk 2]
```diff
@@ -352,9 +353,15 @@ def num_examples(self):
         :rtype: int
         """
         if ClassesMixin._does_quack(self):
-            return len(self.local_importance_values[0])
+            if sp.sparse.issparse(self.local_importance_values[0]):
```
Comment (wamartin-aml) [inline python/interpret_community/explanation/explanation.py]: I don't think there's a better way but this is kinda gross

File: python/interpret_community/explanation/explanation.py (hunk 1 above)
Comment (imatiach-msft) [reply python/interpret_community/explanation/explanation.py]: good point, I added a property that determines if the local importance values are sparse.  I think that makes the code cleaner.

File: python/interpret_community/explanation/explanation.py (hunk 2 above)
Comment (imatiach-msft) [reply python/interpret_community/explanation/explanation.py]: I think I've made it a bit more uniform now, and also more memory efficient in dense case (using local_importance_values doubles memory usage as opposed to _local_importance_values)

**Pull request merged**